pkg/tcpip/link/fdbased/iovlen_unsafe.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package fdbased
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+func setIovLen(h *syscall.Msghdr, iovlen int) {
```

Comment:
I don't think you need to do this in an unsafe way.

I think you can probably have the following:

iovlen_i386.go

// +build i386

func setIovLen(h *syscall.Msghdr, iovlen int) {
   h.Iovlen = uint32(iovlen)
}

iovlen_amd64.go

// +build amd64

func setIovLen(h *syscall.Msghdr, iovlen int) {
   h.Iovlen = uint64(iovlen)
}

Reply:
It's super weird that they have SetControllen and SetLen for Iovecs themselves, but not the relevant method here. They would probably accept a patch upstream also?

Reply:
Yes, the missing Msghdr.SetIovLen() seems like an oversight. I can ask upstream, but the syscall package doc says "Deprecated: this package is locked down.", so I wouldn't hold my breath. There is no SetIovLen() in x/sys/unix either.

I did the unsafe hack to try to catch all 32-bit Linuxes. So iovlen_i386.go, iovlen_amd64.go is not enough, we'd need iovlen_{amd64,arm64,386,arm,mips,...} . But perhaps iovlen_32.go and iovlen_64.go with the appropriate build tags would be best?

```
// iovlen_32.go

// +build linux
// +build 386 arm mips mipsle

package fdbased

func setIovLen(h *syscall.Msghdr, iovlen int) {
    h.Iovlen = uint32(iovlen)
} 
```

```
// iovlen_64.go

// +build linux
// +build arm64 amd64 mips64 mips64le ppc64le ppc64 s390x

package fdbased

func setIovLen(h *syscall.Msghdr, iovlen int) {
    h.Iovlen = uint64(iovlen)
} 
```

Reply:
I updated the PR with iovlen_linux32.go and iovlen_linux64.go without unsafe as suggested above. Also raised an issue about the missing Msghdr.SetIovlen: https://github.com/golang/go/issues/34164